lib/drivers/node-mongodb-native/collection.js

```diff
@@ -77,11 +77,16 @@ function iter(i) {
     const collection = this.collection;
     const args = Array.from(arguments);
     const _this = this;
-    const debug = _this &&
+    const hasGlobalDebug = _this &&
       _this.conn &&
       _this.conn.base &&
       _this.conn.base.options &&
       _this.conn.base.options.debug;
+    const hasConnectionDebug = _this &&
+      _this.conn &&
+      _this.conn.options &&
+      _this.conn.options.debug;
+    const debug = hasGlobalDebug || hasConnectionDebug;
```

Comment:
Check for nullish rather than falsy. So `connectionDebug == null ? globalDebug : connectionDebug` . That's because setting debug to `false` on a connection should override the global debug config.